contracts/interchain-token-service/src/contract/migrations/v1_1_0.rs

```diff
@@ -0,0 +1,134 @@
+use axelar_wasm_std::error::ContractError;
+use axelar_wasm_std::nonempty;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Order, Storage, Uint256, Uint512};
+use cw_storage_plus::Map;
+use itertools::Itertools;
+use router_api::{Address, ChainNameRaw};
+
+use crate::shared::NumBits;
+use crate::state;
+
+#[cw_serde]
+pub struct ChainConfig {
+    pub truncation: TruncationConfig,
+    pub its_address: Address,
+    pub frozen: bool,
+}
+
+#[cw_serde]
+pub struct TruncationConfig {
+    pub max_uint: nonempty::Uint256, // The maximum number of bits used to represent unsigned integer values that is supported by the chain's token standard
+    pub max_decimals_when_truncating: u8, // The maximum number of decimals that is preserved when deploying from a chain with a larger max unsigned integer
+}
+
+const OLD_CHAIN_CONFIGS: Map<&ChainNameRaw, ChainConfig> = Map::new("chain_configs");
+const NEW_CHAIN_CONFIGS: Map<&ChainNameRaw, state::ChainConfig> = Map::new("chain_configs");
+
+pub fn migrate(storage: &mut dyn Storage) -> Result<(), ContractError> {
+    let old_configs: Vec<_> = OLD_CHAIN_CONFIGS
+        .range(storage, None, None, Order::Ascending)
+        .try_collect()?;
+
+    let transformed_configs: Vec<(ChainNameRaw, state::ChainConfig)> = old_configs
+        .into_iter()
+        .map(|(chain_name, old_config)| {
+            convert_max_uint_to_max_bits(old_config.truncation.max_uint).map(|max_uint_bits| {
+                (
+                    chain_name,
+                    state::ChainConfig {
+                        truncation: state::TruncationConfig {
+                            max_decimals_when_truncating: old_config
+                                .truncation
+                                .max_decimals_when_truncating,
+                            max_uint_bits,
+                        },
+                        its_address: old_config.its_address,
+                        frozen: old_config.frozen,
+                    },
+                )
+            })
+        })
+        .try_collect()?;
+
+    for (chain, config) in transformed_configs {
+        NEW_CHAIN_CONFIGS.save(storage, &chain, &config)?;
+    }
+
+    Ok(())
+}
+
+fn convert_max_uint_to_max_bits(max_uint: nonempty::Uint256) -> Result<NumBits, ContractError> {
+    let fixed_cases = [256, 128, 64, 32];
+    for case in fixed_cases {
+        // some chains were incorrectly registered with the max number of bits as the max uint
```

Comment:
Is this migration going to be needed for mainnet?

Reply:
the migration in general or the incorrect number case? In general, yes since the contract was deployed to mainnet. But I don't think any chains were registered incorrectly on mainnet